File: backend/src/Tmap.Api/Features/Notes/NoteDtos.cs
namespace Tmap.Api.Features.Notes;

public sealed record CreateNoteRequest(
    Guid? GroupId,
    Guid? ProjectId,
    string Title,
    string? Content,
    string? Rank = null,
    Guid? Id = null);

public sealed record UpdateNoteRequest(
    Guid? GroupId,
    Guid? ProjectId,
    string? Title,
    string? Content,
    string? Rank);

public sealed record NoteResponse(
    Guid Id,
    Guid? GroupId,
    Guid? ProjectId,
    string Title,
    string Content,
    string Rank,
    DateTimeOffset CreatedAt,
    DateTimeOffset UpdatedAt);

public sealed record ReorderItem(Guid Id, string Rank);

File: android/app/src/main/java/net/qmindtech/tmap/data/remote/dto/NoteDtos.kt
package net.qmindtech.tmap.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class CreateNoteRequest(
    val id: String? = null,
    val groupId: String? = null,
    val projectId: String? = null,
    val title: String,
    val content: String,
    val rank: String? = null,
)

@Serializable
data class UpdateNoteRequest(
    val groupId: String? = null,
    val projectId: String? = null,
    val title: String? = null,
    val content: String? = null,
    val rank: String? = null,
)

@Serializable
data class NoteResponse(
    val id: String,
    val groupId: String?,
    val projectId: String?,
    val title: String,
    val content: String,
    val rank: String?,
    val createdAt: String,
    val updatedAt: String,
)
